emcc.py

```diff
@@ -2224,6 +2230,17 @@ def phase_linker_setup(options, state, newargs):
     if options.use_closure_compiler is None and settings.TRANSPILE_TO_ES5:
       diagnostics.warning('transpile', 'enabling transpilation via closure due to browser version settings.  This warning can be suppressed by passing `--closure=1` or `--closure=0` to opt into our explicitly.')
 
+  # https://caniuse.com/class: EDGE:13 FF:45 CHROME:49 SAFARI:9
+  supports_es6_classes = (settings.MIN_EDGE_VERSION >= 13 and
+                          settings.MIN_FIREFOX_VERSION >= 45 and
+                          settings.MIN_CHROME_VERSION >= 49 and
+                          settings.MIN_SAFARI_VERSION >= 90000 and
+                          settings.MIN_IE_VERSION == 0x7FFFFFFF)
+
+  if not settings.DISABLE_EXCEPTION_CATCHING and settings.EXCEPTION_STACK_TRACES and not supports_es6_classes:
```

Comment:
Hmm.. this makes me thing it would be useful to have an explict way to check for emscripten EH..  `not settings.DISABLE_EXCEPTION_CATCHING` is hard to understand, and also not accurate, since that is also set when wasm exceptions are enabled.

Lets do that as a followup maybe.  This change lgtm.

Reply:
FWIW, Wasm exceptions sets `DISABLE_EXCEPTION_CATCHING` to `1`.
https://github.com/emscripten-core/emscripten/blob/df727461762e0ba77677f95aa4400aae3d50cdfd/emcc.py#L1593-L1603

But indeed, `not settings.DISABLE_EXCEPTION_CATCHING` is a bit difficult, a followup that changes that to `settings.EMSCRIPTEN_EXCEPTIONS` (or something similar) sgtm.

Reply:
Ah, I just noticed that this is simplified in `system_libs.py`.
https://github.com/emscripten-core/emscripten/blob/df727461762e0ba77677f95aa4400aae3d50cdfd/tools/system_libs.py#L740-L745

Perhaps we could move that to `emcc.py` instead?

Reply:
Maybe.. but lets leave that a followup.